src/pint/gridutils.py

```diff
@@ -1,16 +1,400 @@
 """Tools for building chi-squared grids."""
+import concurrent.futures
 import copy
+import logging
 import multiprocessing
 import os
-from multiprocessing import Process, Queue
 
 import astropy.constants as const
 import astropy.units as u
 import numpy as np
 
+try:
+    from tqdm import tqdm
+except ModuleNotFoundError:
+    tqdm = None
+
 import pint.utils
 
-__all__ = ["grid_chisq", "grid_chisq_mp", "plot_grid_chisq"]
+log = logging.getLogger(__name__)
+
+__all__ = ["doonefit", "grid_chisq", "grid_chisq_derived", "plot_grid_chisq"]
+
+
+def doonefit(ftr, parnames, parvalues):
+    """Worker process that computes one fit with specified parameters fixed
+
+    Parameters
+    ----------
+    ftr : pint.fitter.Fitter
+    parnames : list
+        Names of the parameters to grid over
+    parvalues : list
+        List of parameter values to grid over (each should be 1D array of astropy.units.Quantity)
+    """
+    # Make a full copy of the fitter to work with
+    myftr = copy.deepcopy(ftr)
+    for parname, parvalue in zip(parnames, parvalues):
+        # Freeze the  params we are going to grid over and set their values
+        # All other unfrozen parameters will be fitted for at each grid point
+        getattr(myftr.model, parname).frozen = True
+        getattr(myftr.model, parname).quantity = parvalue
+    myftr.fit_toas()
```

Comment:
Do you want to include some kind of handling for fitter failures to converge? Return NaN? Passing errors through parallelization schemes is a messy business. Ideally I suppose this function would pass exceptions upwards and the parallel scheme would use a wrapper that does something more covnenient with them.